From 70bbd8e28c037db834f0b0ad15cde3b2d165f428 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Wed, 30 May 2018 14:41:12 +0300
Subject: [PATCH 0641/1051] fix: irq: ap810: update the start interrupt of ICU
 in the GICv3

On next SoC revision of AP810 (B0), IOMMU context bank interrupt will
be wired to GICv3 entries #96-159.

This patch change the start interrupt of the ICU in the GICv3 to be 160
instad of 96, as preperation for B0.

In addition added note for GICP node:
In Armada-8K-Plus, the CP110 send message to the GICv3,
(instead of AP806 there's GICP unit that recive interrupt
from ICU and send to GIC).

The node of the GICP in AP810 is place holder until we get Marc Z.
patches that allow GICv3 to revice MSI interrupts from irq domain

Change-Id: I1fbc6a3511aea1c144bd03b04046cfae00db303a
Signed-off-by: Hanna Hawa <hannah@marvell.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/arm64/boot/dts/marvell/armada-ap810.dtsi | 6 +++++-
 drivers/irqchip/irq-mvebu-icu.c               | 2 +-
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
index 5d62ef9e0f8a..5f276a6a8b81 100644
--- a/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-ap810.dtsi
@@ -120,10 +120,14 @@
 				status = "disabled";
 			};
 
+			/* TODO: this node is not correct - we should
+			 * remove this node, the ICU should send message
+			 * to the GICv3, there's no GICP in AP810.
+			 */
 			AP810_LABEL(gicp): gicp@3000040 {
 				compatible = "marvell,ap806-gicp";
 				reg = <0x3000040 0x10>;
-				marvell,spi-ranges = <92 64>, <156 64>;
+				marvell,spi-ranges = <160 128>;
 				msi-controller;
 			};
 
diff --git a/drivers/irqchip/irq-mvebu-icu.c b/drivers/irqchip/irq-mvebu-icu.c
index 128ae6031ab1..ad8e509e9192 100644
--- a/drivers/irqchip/irq-mvebu-icu.c
+++ b/drivers/irqchip/irq-mvebu-icu.c
@@ -37,7 +37,7 @@
 #define ICU_SATA1_ICU_ID	107
 
 /* AP810 ICU entries from CP110 are consecutive */
-#define ICU_GIC_SPI_AP810_BASE		92
+#define ICU_GIC_SPI_AP810_BASE		160
 
 struct mvebu_icu {
 	struct irq_chip irq_chip;
-- 
2.17.1

